packages/iTwinUI-react/src/core/Dialog/Dialog.test.tsx

```diff
@@ -42,6 +42,52 @@ it('should pass down the props through DialogContext', async () => {
   expect(onClose).toHaveBeenCalledTimes(2);
 });
 
+it('should have position correctly dependant on viewport', async () => {
+  const dialogContent = (
+    <Dialog.Main>
+      <Dialog.TitleBar titleText='Test title' />
+      <Dialog.Content>Here is my dialog content</Dialog.Content>
+      <Dialog.ButtonBar>
+        <Button styleType='high-visibility'>Confirm</Button>
+        <Button>Close</Button>
+      </Dialog.ButtonBar>
+    </Dialog.Main>
+  );
+
+  const containerViewport = render(
+    <Dialog relativeTo='viewport'>
+      <Dialog.Backdrop />
+      {dialogContent}
+    </Dialog>,
+  );
+
+  const containerContainer = render(
+    <Dialog relativeTo='container'>
+      <Dialog.Backdrop />
+      {dialogContent}
+    </Dialog>,
+  );
+
+  const dialogWrapperViewport = containerViewport.container.querySelector(
+    '.iui-dialog-wrapper',
+  ) as HTMLElement;
+  const dialogWrapperContainer = containerContainer.container.querySelector(
+    '.iui-dialog-wrapper',
+  ) as HTMLElement;
+
+  const backdropViewport = containerViewport.container.querySelector(
+    '.iui-backdrop',
+  ) as HTMLElement;
+  const backdropContainer = containerContainer.container.querySelector(
+    '.iui-backdrop',
+  ) as HTMLElement;
+
+  expect(dialogWrapperViewport).toHaveStyle('position: fixed');
```

Comment:
I added this test to be generic even for the future. Thus the test is named "should have position correctly dependant on viewport", meaning that "all" positioning should be correct, not just backdrop.

Since it is generic, I am testing the style of the dialog (not the backdrop) since that is still supplied with an inline style on [L90-93](https://github.com/iTwin/iTwinUI-react/blob/main/packages/iTwinUI-react/src/core/Dialog/Dialog.tsx#L90-L93)

If this was not generic, we might create a new test when making positioning changes in the future that I felt might not be right, as related test cases should be kept together in one test.